maskit/examples/load_mnist.py

```diff
@@ -0,0 +1,122 @@
+import tensorflow as tf
+import collections
+from sklearn.decomposition import PCA
+from pennylane import numpy as np
+from sklearn.preprocessing import minmax_scale
+
+np.random.seed(42)
+
+MAX_TRAIN_SAMPLES = 11471
+MAX_TEST_SAMPLES = 1952
+
+
+def reduce_image(x):
+    x = np.reshape(x, [1, 28, 28, 1])
+    x = tf.image.resize(x, [4, 4])
+    x = np.reshape(x, [4, 4])
+    return x / 255
+
+
+def remove_contradicting(xs, ys):
+    mapping = collections.defaultdict(set)
+    for x, y in zip(xs, ys):
+        mapping[str(x)].add(y)
+
+    return zip(*((x, y) for x, y in zip(xs, ys) if len(mapping[str(x)]) == 1))
+
+
+def convert_to_binary(x):
+    x_new = []
+    for a in x:
+        c = (a < 0.5).astype(int)
+        c = np.expand_dims(c, axis=0)
+        x_new.append(c)
+
+    x_new = np.concatenate(x_new)
+    print(x_new.shape)
+    return x_new
+
+
+def convert_label(y, classes):
+    # TODO: currently hardcoded to fit with iris dataset
+    # num_classes = len(classes)
+    num_classes = 4
+    a = [0.0 for i in range(num_classes)]
+    for i in range(num_classes):
+        if y == classes[i]:
+            a[i] = 1.0
+            break
```

Comment:
You should be able to do this in one go:
```suggestion
    a = [0.0 for i in range(num_classes)]
    a[num_classes.index(y)] = 1.0
```

Reply:
Assuming that ``y`` is always in ``classes`` or that it's not an error otherwise:
```suggestion
    a = [1.0 if cls == y else 0.0 for cls in classes]
```